platform/default/src/mbgl/platform/thread_pool.cpp

```diff
@@ -54,4 +76,9 @@ void ThreadPool::schedule(std::weak_ptr<Mailbox> mailbox) {
     cv.notify_one();
 }
 
+Scheduler& platform::GetBackgroundScheduler() {
+    static ThreadPool pool(4);
```

Comment:
I think we should consider supporting some way for users of our SDKs to completely clean up after having used a mapview/our SDK, i.e. in this instance (and the similar one further in the PR) could we move the static to file scope, and then have some cleanup function that could deallocate the pool?

I'm thinking of situations where the mapview is just a secondary part of an application, which may be shown temporarily. Being able to completely clean up and have 0 lingering allocations would be nice.

Reply:
Good point. Do we have already some `freeResorces()`-like API in our SDK?

Reply:
Not that I know of, @tmpsantos?

Reply:
We have somethink on these lines on the `Renderer` class:
https://github.com/mapbox/mapbox-gl-native/blob/master/include/mbgl/renderer/renderer.hpp#L58

Reply:
> I'm thinking of situations where the mapview is just a secondary part of an application, which may be shown temporarily. Being able to completely clean up and have 0 lingering allocations would be nice.

That is a good point, I will address this concern.